src/App.js

```diff
@@ -1,5 +1,99 @@
+import { Console, Random } from "@woowacourse/mission-utils";
+import Car from "./Car.js";
+import { validateCarNameList, validateTryCount } from "./validation.js";
+import {
+  CAR_NAME_INPUT_MESSAGE,
+  TRY_COUNT_INPUT_MESSAGE,
+} from "./constants/inputMessage.js";
+import {
+  EMPTY_MESSAGE,
+  EXECUTE_RESULT_MESSAGE,
+  WINNER_MESSAGE,
+} from "./constants/outputMessage.js";
+
 class App {
-  async run() {}
+  #tryCount;
+  #cars;
+
+  constructor() {
+    this.#tryCount = 0;
+    this.#cars = [];
+  }
+
+  async run() {
+    let carNameList = await this.readCarName();
+    validateCarNameList(carNameList);
+    this.generateCars(carNameList);
+
+    const tryCount = await this.readTryCount();
+    validateTryCount(Number(tryCount));
+    this.#tryCount = Number(tryCount);
+
+    this.printGameStart();
+    this.generateGame();
+
+    const winner = this.generateGameResult();
+    this.printGameResult(winner);
+  }
+
+  async readCarName() {
+    return await Console.readLineAsync(CAR_NAME_INPUT_MESSAGE);
+  }
+
+  async readTryCount() {
+    return await Console.readLineAsync(TRY_COUNT_INPUT_MESSAGE);
+  }
+
+  generateCars(carNameListProp) {
+    const carNameList = carNameListProp.split(",");
+    carNameList.forEach((elem) => this.#cars.push(new Car(elem)));
+  }
+
+  printGameStart() {
+    Console.print(EXECUTE_RESULT_MESSAGE);
+  }
+
+  generateGame() {
+    for (let i = 0; i < this.#tryCount; i++) {
+      this.generateGamePerCycle();
+      Console.print(EMPTY_MESSAGE);
+    }
+  }
+
+  generateGamePerCycle() {
+    this.#cars.forEach((car) => {
+      const randomNumber = this.calculateRandomNumber();
+      this.generateCarMove(car, randomNumber);
+      this.printCarmove(car);
+    });
+  }
+
+  generateCarMove(car, randomNumber) {
+    if (randomNumber >= 4) {
+      car.increaseProgressCount();
+    }
+  }
+
+  calculateRandomNumber() {
+    return Random.pickNumberInRange(0, 9);
+  }
```

Comment:
```suggestion
  calculateRandomNumber(min, max) {
    return Random.pickNumberInRange(min, max);
  }
```
랜덤 숫자의 범위를 인자로 받으면 추후 요구사항이 변경되어도 유연하게 동작할 수 있을 것 같아요~!

Reply:
유지보수에 도움이 되는 코드군요~! 
상수를 하드코딩하지 않고 변수로 빼주는 전략을 이곳에서 놓쳤네요!
감사합니다!